compiler/rustc_span/src/lib.rs

```diff
@@ -1318,17 +1318,26 @@ impl<D: Decoder> Decodable<D> for SourceFile {
                 let mut line_start: BytePos = Decodable::decode(d);
                 lines.push(line_start);
 
-                for _ in 1..num_lines {
-                    let diff = match bytes_per_diff {
-                        1 => d.read_u8() as u32,
-                        2 => d.read_u16() as u32,
-                        4 => d.read_u32(),
-                        _ => unreachable!(),
-                    };
-
-                    line_start = line_start + BytePos(diff);
-
-                    lines.push(line_start);
+                match bytes_per_diff {
+                    1 => {
+                        for _ in 1..num_lines {
+                            line_start = line_start + BytePos(d.read_u8() as u32);
+                            lines.push(line_start);
```

Comment:
What about using `extend` with an iterator instead of `push`?

Reply:
You mean like this?
```
lines.extend((1..num_lines).map(|_| {
    line_start += BytePos(d.read_u8());
    line_start
}));
```
It's harder to read but could be faster if it avoids bounds checks. @martingms , want to try it out?

Reply:
Yes, like that 👍

Reply:
I decided against it for readability, but I didn't benchmark it to see if it's faster, will do that now.

Reply:
Pushed the commit here, if you wanted to do a CI perf run @nnethercote

Reply:
Looks promising locally, so worth a run! Thanks for the suggestion @Dandandan, I've still got a lot to learn about convincing the compiler to elide bounds checks.